src/test/java/com/google/api/codegen/protoannotations/testdata/library.proto

```diff
@@ -427,8 +519,7 @@ message Shelf {
 message CreateShelfRequest {
   // The shelf to create.
   Shelf shelf = 1 [
-    (google.api.required) = true,
-    (google.api.resource_type) = "google.example.library.v1.Shelf"];
```

Comment:
There's probably something I don't understand. Does the current GAPIC YAML not use resource names for these?

Reply:
The `resource_type` annotation only makes sense for `string` type fields. This is a weird circular reference, where the `Shelf shelf` refers to a `resource_type = "google.example.library.v1.Shelf"`, which doesn't provide any new information. I had previously put this annotation in to match what the annotated pubsub.proto used to look like, but those changes were recently reverted in https://github.com/googleapis/googleapis/pull/520

Reply:
Oh right I get it now.